openedx/core/lib/cache_utils.py

```diff
@@ -6,40 +6,101 @@
+          args/kwargs that can't be converted to strings, you're gonna have a bad time (don't do it)
+        - Cache key cardinality depends on the args/kwargs, so if you're caching a function that takes five arguments,
+          you might have deceptively low cache efficiency.  Prefer function with fewer arguments.
+        - We use the default request cache, not a named request cache. The code automatically namespaces the cache
+          key with the module and function's name.
+        - If you require a named request cache, use the ns_request_cached decorator below. Generally, you would need it
+          only if you need control of your own namespaced cache - for example, to clear out your own cache.
+        - WATCH OUT: Don't use this decorator for instance methods that take in a "self" argument that changes each
+          time the method is called. This will result in constant cache misses and not provide the performance benefit
+          you are looking for. Rather, change your instance method to a class method.
+        - Benchmark, benchmark, benchmark! if you never measure, how will you know you've improved? or regressed?
 
     Arguments:
-        request_cache_attr_name - The name of the field or property in this method's containing
-         class that stores the request_cache.
+        f (func): the function to wrap
+
+    Returns:
+        func: a wrapper function which will call the wrapped function, passing in the same args/kwargs,
+              cache the value it returns, and return that cached value for subsequent calls with the
+              same args/kwargs within a single request
+    """
+    return ns_request_cached()(f)
+
+
+def ns_request_cached(namespace=None, arg_map_function=None, request_cache_getter=None):
     """
-    def _decorator(func):
-        """Outer method decorator."""
-        @functools.wraps(func)
-        def _wrapper(self, *args, **kwargs):
+    Same as request_cached above, except an optional namespace can be passed in to compartmentalize the cache.
+
+    Arguments:
+        namespace (string): An optional namespace to use for the cache.  Useful if the caller wants to manage
+            their own sub-cache by, for example, calling RequestCache(namespace=NAMESPACE).clear() for their own
+            namespace.
+        arg_map_function (function: arg->string): Function to use for mapping the wrapped function's arguments to
+            strings to use in the cache key. If not provided, defaults to force_text, which converts the given
+            argument to a string.
+        request_cache_getter (function: args->RequestCache): Function that returns the RequestCache to use. If not
+            provided, defaults to edx_django_utils.cache.RequestCache.  If None, the function's return values are
+            not cached.
+    """
+    def outer_wrapper(f):
+        """
+        Outer wrapper that decorates the given function
```

Comment:
Can you give more clarity in the docstrings as to the difference between `outer_wrapper` and `inner_wrapper`?

Reply:
Once again, this is pre-existing code that I moved here.  But yes, let me see what I can do in clarifying the docstrings.